src/Razor/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/DefaultDocumentWriter.cs

```diff
@@ -131,7 +131,19 @@ public override void VisitDocument(DocumentIntermediateNode node)
 
             public override void VisitUsingDirective(UsingDirectiveIntermediateNode node)
             {
+                var isDefault = !node.Source.HasValue || (node.Source.HasValue && node.Source.Value.FilePath is null);
+                var isImportFile = node.Source.HasValue && node.Source.Value.FilePath != Context.SourceDocument.FilePath;
+                var isExternalImport = isDefault || isImportFile;
```

Comment:
Slightly confusing what's an "externalImport" represent?

Reply:
Something is an `externalImport` if the 'source' of the using statement is something other than the document we're generating C# for.